smtlib-backends-process/src/SMTLIB/Backends/Process.hs

```diff
@@ -28,20 +28,22 @@ import Data.Default (Default, def)
 import GHC.IO.Exception (IOException (ioe_description))
 import SMTLIB.Backends (Backend (..))
 import qualified System.IO as IO
-import System.Process as P
+import qualified System.Process as P
 
 data Config = Config
   { -- | The command to call to run the solver.
     exe :: String,
     -- | Arguments to pass to the solver's command.
-    args :: [String]
+    args :: [String],
+    -- | How to handle std_err of the solver.
+    std_err :: P.StdStream
   }
 
 -- | By default, use Z3 as an external process and ignores log messages.
 defaultConfig :: Config
 -- if you change this, make sure to also update the comment two lines above
 -- as well as the one in @smtlib-backends-process/tests/Examples.hs@
-defaultConfig = Config "z3" ["-in"]
+defaultConfig = Config "z3" ["-in"] P.CreatePipe
```

Comment:
```suggestion
defaultConfig = Config "z3" ["-in"] P.NoStream
```
Shouldn't the error channel be created only if the user needs it ?

Reply:
This would require changing the `underlyingProcess` example as well.

Reply:
The documentation of Process says processes could fail when writing to `stderr` if we do this. For that reason, I figured that `CreatePipe` could be a better default. If the process is writing an error, it probably should fail anyway, but if the process is writing a warning, then it could be problematic to have it fail.